Update `Combobox` read.me documentation (#5321)
Shopify/polaris: polaris-react/src/components/Combobox/README.md

# Combobox

> Combobox uses an enahnced text field that allows merchants to filter through a list of options to pick one or more values. The list of options is displayed when a merchant focuses on the field.

Reviewer: Since this is only true in some instances (single select, multi-select with previously selected options), should the last sentence be removed and left to the single vs multi select content to explain?

> Combobox uses an enhanced text field that allows merchants to filter through a list of options to pick one or more values. The list of options is displayed when a merchant focuses on the field.

Reviewer: ```suggestion
Combobox is an accessible autocomplete input that enables merchants to filter a list of options and select one or more values.
```

---

## Anatomy

![A diagram of the Combobox component showing the smaller primitive components it is composed of.](/public_images/components/Combobox/combobox-anatomy.png)

A combobox is made up of the following:

> 1. **Text field**: The field people click in to activate the popover and filter through the options they can choose from. Once selected, the option will be shown in the text input.

Reviewer: ```suggestion
1. **TextField**: A text input that activates a popover displaying a list of options. As merchants type in the text field, the list of options is filtered by the input value. Options replace or add to the input value when selected.
```

> 2. **Popover**: Renders to contain the Listbox.

Reviewer: ```suggestion
2. **Popover**: An overlay containing a list of options.
```

> 3. **Listbox**: Contains the list of all selectable items.

Reviewer: ```suggestion
3. **Listbox**: A list of options to filter and select or deselect.
```

> 4. **Listbox** options: The individual options that merchants can select. For context on ways the Listbox can be composed with various content, check out the [listbox documentation](https://polaris.shopify.com/components/forms/listbox).

Reviewer: ```suggestion
4. **Listbox.Option**: The individual options to select or deselect. Check out the [listbox component documentation](https://polaris.shopify.com/components/forms/listbox) to learn how to compose it with various content.
```

---

## Best practices

The `Combobox` component should:

> - Be clearly labeled so the merchant knows what type of options will be available

Reviewer: ```suggestion
- Be clearly labeled so the merchant knows what kind of options will be available
```

---

## Sorting and filtering

### Sorting

Item order should be intentional. Order them so it’s easy for the merchant to find a specific value. Some ways you can do this:

- Sort options in alphabetical order
- Display options based on how frequently the merchant selects an option

If multiple options can be selected, move selected items to the top of the list. If this doesn’t work for your context, you can override this behavior.

### Filtering

- By default, menu items are filtered based on whether or not they match the value of the textfield.
- Filters are **not** case-sensitive by default.
- You can apply custom filtering logic if the default behavior doesn’t make sense for your use case.

---

## Patterns

### Tags autocomplete

> The tag multiselect allows merchants to select, create, and browse from a long list of options.

Reviewer: ```suggestion
The tag multi-select input enables merchants to efficiently add or remove tags from a resource, like a product or an order. It uses the inline autocomplete combobox pattern to present merchants with an editable list of tags to browse and select from.
```

---

## Examples

### Single select autocomplete

> Allows merchants to select from a predefined list of options. It’s typically used when there are a large number of options to choose from.

Reviewer: ```suggestion
Use when merchants can select one option from a predefined or editable list..
```

### Multi-select autocomplete

> Allows the merchant to select multiple options from a pre-defined list of options.

Reviewer: ```suggestion
Use when merchants can select one or more options from a predefined or editable list.
```
